assets/pages/new-contributions-landing/components/ContributionForm.jsx

```diff
@@ -0,0 +1,202 @@
+// @flow
+
+// ----- Imports ----- //
+
+import React from 'react';
+import { connect } from 'react-redux';
+import { Redirect } from 'react-router';
+import type { Dispatch } from 'redux';
+
+import { countryGroupSpecificDetails, type CountryMetaData } from 'helpers/internationalisation/contributions';
+import { type CountryGroupId } from 'helpers/internationalisation/countryGroup';
+import { classNameWithModifiers } from 'helpers/utilities';
+import { type ReferrerAcquisitionData } from 'helpers/tracking/acquisitions';
+import { type OptimizeExperiments } from 'helpers/tracking/optimize';
+import { type IsoCurrency } from 'helpers/internationalisation/currency';
+import { type Participations } from 'helpers/abTests/abtest';
+import { setupStripeCheckout, openDialogBox, createTokenCallback } from 'helpers/paymentIntegrations/stripeCheckout';
+import trackConversion from 'helpers/tracking/conversions';
+
+import ErrorMessage from 'components/errorMessage/errorMessage';
+import SvgEnvelope from 'components/svgs/envelope';
+import SvgUser from 'components/svgs/user';
+
+import { NewContributionType } from './ContributionType';
+import { NewContributionAmount } from './ContributionAmount';
+import { NewContributionPayment } from './ContributionPayment';
+import { NewContributionState } from './ContributionState';
+import { NewContributionSubmit } from './ContributionSubmit';
+import { NewContributionTextInput } from './ContributionTextInput';
+
+import { type State } from '../contributionsLandingReducer';
+import { type Action, paymentSuccess, paymentFailure } from '../contributionsLandingActions';
+
+// ----- Types ----- //
+/* eslint-disable react/no-unused-prop-types */
+type PropTypes = {|
+  done: boolean,
+  error: string | null,
+  isTestUser: boolean,
+  countryGroupId: CountryGroupId,
+  currency: IsoCurrency,
+  selectedCountryGroupDetails: CountryMetaData,
+  abParticipations: Participations,
+  referrerAcquisitionData: ReferrerAcquisitionData,
+  optimizeExperiments: OptimizeExperiments,
+  thankYouRoute: string,
+  onSuccess: () => void,
+  onError: string => void,
+|};
+/* eslint-enable react/no-unused-prop-types */
+
+const mapStateToProps = (state: State) => ({
```

Comment:
You can read the values for first name, last name, email etc from the user state here. 

So something like:

```
const mapStateToProps = (state: State) => ({
  done: state.page.form.done,
  isTestUser: state.page.user.isTestUser || false,
  referrerAcquisitionData: state.common.referrerAcquisitionData,
  abParticipations: state.common.abParticipations,
  optimizeExperiments: state.common.optimizeExperiments,
  firstName: state.page.user.firstName,
  lastName: state.page.user.lastName,
  email: state.page.user.email,
});
```

That way you don't need to read from the `window.guardian` object further down